scripts/src/javatests/org/oppia/android/scripts/xml/XmlSyntaxCheckTest.kt

```diff
@@ -0,0 +1,128 @@
+package org.oppia.android.scripts.xml
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.scripts.common.XML_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR
+import org.oppia.android.scripts.common.XML_SYNTAX_CHECK_PASSED_OUTPUT_INDICATOR
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [XmlSyntaxCheck]. */
+class XmlSyntaxCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("testfiles")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testXmlSyntax_validXml_xmlSyntaxIsCorrect() {
+    val validXml =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <shape xmlns:android="http://schemas.android.com/apk/res/android"
+        android:shape="rectangle">
+        <solid android:color="#3333334D" />
+        <size android:height="1dp" />
+      </shape>
+      """.trimIndent()
+
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXml)
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(XML_SYNTAX_CHECK_PASSED_OUTPUT_INDICATOR)
+  }
+
+  @Test
+  fun testXmlSyntax_invalidOpeningTag_xmlSyntaxIsIncorrect() {
+    val invalidXml =
+      """
+      <?xml version="1.0" encoding="utf-8"?>
+      <shape xmlns:android="http://schemas.android.com/apk/res/android"
+        android:shape="rectangle">
+        <solid android:color="#3333334D" />
+        <size android:height="1dp" />
+      </shapes>
+      """.trimIndent()
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXml)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(XML_SYNTAX_CHECK_FAILED_OUTPUT_INDICATOR)
+    assertThat(outContent.toString().trim()).isEqualTo(
+      """
+      ${retrieveTestFilesDirectoryPath()}/TestFile.xml:6:8: The end-tag for element type "shape" must end with a '>' delimiter.
```

Comment:
Here & elsewhere: we should try to avoid exceeding the column limit, and in this case I think it's avoidable. Consider changing this to:

```kotlin
assertThat(outContent.toString()).contains("${retrieveTestFilesDirectoryPath()}/TestFile1.xml:6:8: The end-tag")
assertThat(outContent.toString()).contains("The end-tag for element type "shape" must end with a '>' delimiter.")
```

In this case I don't think it's important to necessarily verify the exact error since that's dependent on the XML parser, but it is important to verify the formatting & presence of the error since that's influenced by code we maintain.

Reply:
Revamped the tests to check output format, pass status and failures.